R/as_tibble.R

```diff
@@ -110,46 +112,40 @@ as_tibble.data.frame <- function(x, validate = TRUE, ...,
 #' @export
 #' @rdname as_tibble
 as_tibble.list <- function(x, validate = TRUE, ..., .rows = NULL,
-                           .tidy_names = if (validate) NULL else FALSE) {
+                           .name_repair = NULL) {
   if (!missing(validate)) {
-    warn("The `validate` argument to `as_tibble()` is deprecated. Please use `.tidy_names` to control column names.")
+    warn("The `validate` argument to `as_tibble()` is deprecated. Please use `.name_repair` to control column names.")
+    .name_repair <- if (isTRUE(validate)) "none_passive" else "none"
   }
 
-  x <- auto_tidy_names(x, .tidy_names)
-  recycle_columns(x, .rows)
-}
+  .name_repair <- .name_repair %||% "none_passive"
 
-auto_tidy_names <- function(x, .tidy_names) {
-  if (is.null(.tidy_names)) {
-    x <- check_tibble(x)
-  } else if (isTRUE(.tidy_names)) {
-    names(x) <- tidy_names(names2(x))
-  } else if (is_function(.tidy_names)) {
-    names(x) <- .tidy_names(names2(x))
-  } else if (identical(.tidy_names, FALSE)) {
-    if (has_null_names(x)) {
-      names(x) <- rlang::rep_along(x, "")
-    }
-  } else {
-    abort(error_tidy_names_arg())
-  }
-  x
-}
+  x <- set_minimal_names(x)
 
-check_tibble <- function(x) {
-  # Names
-  names_x <- names2(x)
-  bad_name <- which(is.na(names_x) | names_x == "")
-  if (has_length(bad_name)) {
-    abort(error_column_must_be_named(bad_name))
+  if (is_function(.name_repair)) {
```

Comment:
Perhaps I should extract the all name repair below here into a function defined in repair-names.R, so this logic isn't locked up in `as_tibble.list`.

Reply:
Yes, I think that would be a good idea. That would also mean that we don't have to expose `valid_names()` etc; otherwise we'll need to spend some time discussing a good naming strategy for those functions (i.e. we now have a family of functions where the family name is a suffix, not a prefix, which is undesirable)

Reply:
I know the names aren't great. The existence `tidy_names()` and `set_tidy_names()` must be reckoned with, when we do tackle it.